src/main/java/games/strategy/triplea/ui/MovePanel.java

```diff
@@ -1142,16 +1127,13 @@ private void selectEndPoint(final Territory territory) {
         }
         // this match will make sure we can't select more units
         // of a specific type then we had originally selected
-        final Match<Collection<Unit>> unitTypeCountMatch = new Match<Collection<Unit>>() {
-          @Override
-          public boolean match(final Collection<Unit> units) {
-            final IntegerMap<UnitType> currentMap = new IntegerMap<>();
-            for (final Unit unit : units) {
-              currentMap.add(unit.getType(), 1);
-            }
-            return maxMap.greaterThanOrEqualTo(currentMap);
+        final Match<Collection<Unit>> unitTypeCountMatch = Match.of(unitsToCheck -> {
```

Comment:
I renamed `units` to `unitsToCheck` to avoid a local variable name conflict due to the conversion from an anonymous class to a lambda.